### DISCUSSION TITLE
'HardwareSerial' was not declared in this scope

### DISCUSSION THREAD

**Discussion by spectraljules** [FortySevenEffects/arduino_midi_library#282] [Q&A] [answered]
EDIT: just realized that Attiny 84a doesn't have hardware serial.  I read the information on it wrong.  D'oh!  

Decided to start my project over with an Attiny 84a rather than an 85 so I don't have to mess with softserial.  Now when I try to compile I get 'HardwareSerial' was not declared in this scope.  

> 
#include <MIDI.h>

MIDI_CREATE_DEFAULT_INSTANCE();

void setup() {
  pinMode(2, OUTPUT);
}

void loop() {
  digitalWrite(2, HIGH);
  delay(500);
  digitalWrite(2, LOW);
  delay(500);
}

**Comment by franky47** [comment, marked as answer]
From what I can read in the [Arduino core code](https://github.com/arduino/ArduinoCore-avr/blob/bc792a9a716289249fa5f2f67be3ee8ba6507d6c/cores/arduino/HardwareSerial.cpp#L36-L37), HardwareSerial is disabled for ATtiny chips, which only have a USI (Universal Serial Interface), which does not use the same kind of registers as a dedicated UART, which is what the HardwareSerial interface relies on.
